src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1613,7 +1613,11 @@ def _execute_command(self, **kwds):
 
     def __call__(self, **kwds):
         """Call a command with the specified keyword arguments."""
-        return self.execute_command(**kwds)
+        try:
+            return self.execute_command(**kwds)
+        except KeyboardInterrupt:
+            if self.obj_name in ["iterate", "calculate", "dual_time_iterate"]:
```

Comment:
This list of names is too concrete for a permanent solution.

Reply:
We can actually change it to child names of "run_calculation".

Reply:
Which issue does that resolve?

Reply:
As discussed, we propose to check if parent has an interrupt method such that interrupt is explicitly a concept in the framework.

Reply:
@mkundu1 do you thoughts on this topic?

Reply:
I think as `interrupt` is specific command in the settings API, it is better to call it explicitly `solver.settings.solution.run_calculation.interrupt()`. We can wrap it within a top-level method like `solver.interrupt()` (can be private).

Detecting interruptible commands through a settings API attribute is a good idea, we need to check with the settings API team.

Reply:
Thanks @mkundu1. We had touched on this and I think this is the best plan.

Reply:
FYI @mkundu1 @seanpearsonuk there is an `interruptible?` attribute in the Settings API (and in the Rest API meshing datamodel wrapper), we make use of it in the WebUI